src/org/labkey/test/components/react/MultiMenu.java

```diff
@@ -186,34 +206,182 @@ public List<String> getMenuText()
+                return item;
+        }
+        throw new NoSuchElementException("No item '" + label + "' under toggle '" + toggleLabel + "' found.");
+    }
 
-        return items;
+    /**
+     * Clicks the specified item under the specified heading
+     *
+     * @param heading heading under which to find the item
+     * @param label   text of the item
+     */
+    public void clickItemUnderHeading(String heading, String label)
+    {
+        WebElement item = getItemUnderHeading(heading, label);
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(item));
+        item.click();
+    }
+
+    /**
+     * gets the names of dropdown-section__menu-items between a heading and the next divider
+     *
+     * @param heading The text of the heading
+     * @param collapse Whether to collapse the menu when finished
+     * @return
+     */
+    public List<String> getItemsUnderToggle(String heading, boolean collapse)
+    {
+        var itemTexts = getWrapper().getTexts(getItemsUnderToggle(heading));
+        if (collapse)
+            collapse();
+        return itemTexts;
+    }
+
+    /**
+     * Gets a list of dropdown-section__menu-items under a given toggle. For cases where the menu has duplicate
+     * menu-items and you need to select one under a specific toggle, use this to find it
+     *
+     * @param label Text of the toggle item
+     * @return A list of menu items under the specified toggle.
+     */
+    protected List<WebElement> getItemsUnderToggle(String label)
+    {
+        expand();
+        expandToggle(label);
+        waitForData();
+
+        List<WebElement> itemsUnderToggle = new ArrayList<>();
+
+        // iterate from the top and generate the list of items that follow it
+        List<WebElement> listItems = getListItems();
+        boolean headingFound = false;
+
+        for (WebElement item : listItems)
+        {
+            String className = item.getAttribute("class");
+            String text = item.getText().trim();
+
+            if (className.contains("dropdown-section-toggle") && text.equalsIgnoreCase(label))
+                headingFound = true;
+
+            // Once we've found our toggle we know that all dropdown-section__menu-item elements belong to the heading
+            // we are interested in
+            if (headingFound && className.contains("dropdown-section__menu-item"))
+                itemsUnderToggle.add(item);
+
+            // Once we hit another divider or another toggle we're done looking at menu items related to the toggle, so we can stop iterating
+            if (headingFound && !text.equalsIgnoreCase(label) && className.contains("dropdown-section-toggle"))
+                break;
+        }
+
+        return itemsUnderToggle;
+    }
+
+    private WebElement getItemUnderToggle(String toggleLabel, String label)
+    {
+        waitForData();
+        var itemsUnderToggle = getItemsUnderToggle(toggleLabel);
```

Comment:
`getItemsUnderToggle` already waits for data
```suggestion
        var itemsUnderToggle = getItemsUnderToggle(toggleLabel);
```

Reply:
Thank you, good catch

Reply:
I'm now considering whether to put waitForData into the path for getting first-level menu items as well